File: vortexdbg-android/src/main/java/com/vortexdbg/linux/struct/Dirent.java
package com.vortexdbg.linux.struct;

import com.vortexdbg.pointer.VortexdbgStructure;
import com.sun.jna.Pointer;

import java.util.Arrays;
import java.util.List;

public class Dirent extends VortexdbgStructure {

    public Dirent(Pointer p) {
        super(p);
    }

    public long d_ino;
    public long d_off;
    public short d_reclen;
    public byte d_type;
    public byte[] d_name = new byte[256];

    @Override
    protected List<String> getFieldOrder() {
        return Arrays.asList("d_ino", "d_off", "d_reclen", "d_type", "d_name");
    }

}

File: vortexdbg-android/src/main/kotlin/com/vortexdbg/linux/struct/Dirent.kt
package com.vortexdbg.linux.struct

import com.vortexdbg.pointer.VortexdbgStructure
import com.sun.jna.Pointer

class Dirent(p: Pointer?) : VortexdbgStructure(p) {

    @JvmField
    var d_ino: Long = 0
    @JvmField
    var d_off: Long = 0
    @JvmField
    var d_reclen: Short = 0
    @JvmField
    var d_type: Byte = 0
    @JvmField
    var d_name = ByteArray(256)

    override fun getFieldOrder(): List<String> {
        return listOf("d_ino", "d_off", "d_reclen", "d_type", "d_name")
    }

}
